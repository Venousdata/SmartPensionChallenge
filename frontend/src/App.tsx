import React, { useEffect, useState } from "react";
import "./App.css";
import { post } from "./api/api";
import {
  Table,
  TableContainer,
  TableHead,
  TableRow,
  TableCell,
  TableBody,
  Paper,
} from "@material-ui/core";

export default function App() {
  const [sortBy, setSortBy] = useState("view");
  const [ascDesc, setAscDesc] = useState(false);
  const [tableRows, setTableRows]: any[] = useState([]);
  const tableHeaders = ["Path", "Views", "Unique Views"];

  useEffect(() => {
    post("webpageviews", { sortBy, order: ascDesc ? "asc" : "desc" }).then(
      (data) => {
        setTableRows(data || []);
      }
    );
  }, [sortBy, ascDesc]);

  const sortByHeader = (header: string, ascDesc: boolean) => {
    const headerMap: { [key: string]: string } = {
      Path: "path",
      Views: "view",
      "Unique Views": "uniqueView",
    };
    setSortBy(headerMap[header]);
    setAscDesc(ascDesc);
  };

  return (
    <div className="App">
      <TableContainer component={Paper}>
        <Table style={{ minWidth: 650 }}>
          <TableHead>
            <TableRow>
              {tableHeaders.map((header, idx) => (
                <TableCell key={idx}>
                  {header}
                  {header !== "Path" && (
                    <button
                      key={idx}
                      onClick={() => sortByHeader(header, !ascDesc)}
                    >
                      ascDesc
                    </button>
                  )}
                </TableCell>
              ))}
            </TableRow>
          </TableHead>
          <TableBody>
            {tableRows.map((row: any, idx: number) => (
              <TableRow key={idx} style={{ border: 0 }}>
                {Object.keys(row).map((key: string) => (
                  <TableCell>{row[key]}</TableCell>
                ))}
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </div>
  );
}
