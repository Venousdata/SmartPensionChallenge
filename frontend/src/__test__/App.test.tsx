/**
 * @jest-environment jsdom
 */

import React from "react";
import { render } from "@testing-library/react";
import App from "../App";

describe("App tests", () => {
  it("App should render and match snapshot", (done) => {
    const container = render(<App />);
    expect(container).toMatchSnapshot();
    done();
  });
});
