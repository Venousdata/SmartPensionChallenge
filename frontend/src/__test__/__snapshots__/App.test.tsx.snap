// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests App should render and match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <table
            class="MuiTable-root"
            style="min-width: 650px;"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Path
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Views
                  <button>
                    ascDesc
                  </button>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Unique Views
                  <button>
                    ascDesc
                  </button>
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            />
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <table
          class="MuiTable-root"
          style="min-width: 650px;"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Path
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Views
                <button>
                  ascDesc
                </button>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Unique Views
                <button>
                  ascDesc
                </button>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          />
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
